### PR TITLE
Add bookmark delete UI, and edit bookmark list function

### LINKED ISSUE #171: [PE-D] Suggestion for Users to be able to delete or search Bookmarks stored
There do not seem to be a way to perform operations on bookmarks, such as delete. Perhaps it is to be implemented by the final milestone, by which I would recommend some form of search and delete operations for users to manage bookmarks.

-------------
Labels: `severity.Low` `type.FeatureFlaw`
original: kaiwen98/ped#2

### PR DESCRIPTION
**Pull request by josephhhhhhhhh**
to show the correct answer in bookmarked questions. Closes #171 .

### CODE REVIEW

**Comment by kstonekuan** [comment]
Looks good, can you help add some screenshots to the UG as well as placeholders before the final ones?

**Comment by Zhi-You** [comment]
In addition to screenshots in UG, I believe a Parser diagram's link is broken in DG too. Could you check what is wrong with that too? Thanks!

File: src/main/java/seedu/eduke8/ui/Ui.java [hunk 1]
```diff
@@ -492,6 +495,12 @@ public void printListOfBookmarkedQuestions(BookmarkList bookmarks) {
         printMessage(MESSAGE_BOOKMARK_LIST + System.lineSeparator() + list);
     }
 
+    public void printDeletedBookmarkIndicator(int deleteIndex) {
```
Comment (joelngyx) [inline src/main/java/seedu/eduke8/ui/Ui.java:536]: Shouldn't variables be named using nouns and note with a verb? A better way of naming `deleteIndex` could be `indexToDelete`.

File: src/main/java/seedu/eduke8/ui/Ui.java [hunk 2]
```diff
@@ -503,7 +512,9 @@ private String listOfBookmarkedQuestions(BookmarkList bookmarks) {
             ArrayList<Displayable> optionsAvailable = properQuestion.getOptionList().getInnerList();
             int j = 1;
             for (Displayable option : optionsAvailable) {
+                Option castedOption = (Option) option;
                 optionOutput += System.lineSeparator() + "    " + j + CLOSE_BRACKET + option.getDescription()
+                        + (castedOption.isCorrectAnswer() ? MESSAGE_BOOKMARK_CORRECT_ANS_INDICATOR : "")
```
Comment (joelngyx) [inline src/main/java/seedu/eduke8/ui/Ui.java:555]: Perhaps a constant could be used in place of `""`

**Review by joelngyx — COMMENTED** [review]
Code looks sound, but there are some potential problems with code convention!

File: src/main/java/seedu/eduke8/parser/MenuParser.java [hunk 3]
```diff
@@ -161,6 +162,11 @@ public Command parseCommand(DisplayableList topicList, String userInput) {
             return new QuizCommand((TopicList) topicList, numOfQuestions, topicName, ui, bookmarks, userTimer);
         case COMMAND_BOOKMARK:
             LOGGER.log(Level.INFO, "Parsing complete: bookmark command chosen.");
+            if (commandArr.length == 3) {
```
Comment (anqi20) [inline src/main/java/seedu/eduke8/parser/MenuParser.java]: Take note of the magic number!

**Review by anqi20 — APPROVED** [review]
There's a few magic literals and numbers but it's alright I will do a last check through when everything is being merged in :) 
Maybe it would be good to see if you want to take in Joel's comment

File: src/main/java/seedu/eduke8/ui/Ui.java (hunk 2 above)
Comment (josephhhhhhhhh) [reply src/main/java/seedu/eduke8/ui/Ui.java:555]: This one I'm not too sure, I noticed other classes with blank strings as well, figured we were leaving empty strings as empty strings.

File: src/main/java/seedu/eduke8/parser/MenuParser.java (hunk 3 above)
Comment (josephhhhhhhhh) [reply src/main/java/seedu/eduke8/parser/MenuParser.java]: Done. Thanks!

**Pull request merged**